Gradle `:hedera-node:app:run` task broken on `develop`

### Description

Previously we were able to run mono-service locally via `:hedera-node:app:run`.

Now there are failure modes including (for various developers),
```
Caused by: com.swirlds.platform.config.legacy.ConfigurationException: ERROR: Configuration file not found: /Users/michaeltinker/Dev/hedera-services/hedera-node/hedera-app/build/node/config.txt
```

and,
```
Execution failed for task ':app-service-util-impl:compileJava'.
> Could not resolve all files for configuration ':app-service-util-impl:compileClasspath'.
   > Failed to transform app-service-util-0.45.0-SNAPSHOT.jar (project :app-service-util) to match attributes {artifactType=jar, javaModule=true, org.gradle.category=library, org.gradle.dependency.bundling=external, org.gradle.jvm.version=17, org.gradle.libraryelements=jar, org.gradle.usage=java-api}.
      > Execution failed for ExtraJavaModuleInfoTransform: /Users/neeharikasompalli/Documents/Hedera/Repos/repo2/hedera-services/hedera-node/hedera-util-service/build/libs/app-service-util-0.45.0-SNAPSHOT.jar.
         > Not a module and no mapping defined: app-service-util-0.45.0-SNAPSHOT.jar
```

### Steps to reproduce

Checkout `develop`.

Run `:hedera-node:app:run`

### Additional context

_No response_

### Hedera network

other

### Version

`develop`

### Operating system

None

--- a/hedera-node/hedera-app/build.gradle.kts
+++ b/hedera-node/hedera-app/build.gradle.kts
@@ -141,32 +141,6 @@ tasks.jar {
     }
 }
 
-// Define build directory
-val buildDir = layout.projectDirectory.dir("./build/node")
-
-// Copy everything from hedera-node/data
-val copyNodeData =
-    tasks.register<Sync>("copyNodeData") {
-        dependsOn(copyLib)
-        dependsOn(copyApp)
-        from(layout.projectDirectory.dir("../data/keys")) { into("data/keys") }
-        from(layout.projectDirectory.dir("../data"))
-        into(buildDir)
-        exclude("config", "keys") // Exclude config directory
-        shouldRunAfter(tasks.named("copyApp"))
-        shouldRunAfter(tasks.named("copyLib"))
-    }
-
-//// Copy hedera-node/configuration/dev as hedera-node/hedera-app/build/node/data/config  }
-val copyConfig =
-    tasks.register<Copy>("copyConfig") {
-        from(layout.projectDirectory.dir("../configuration/dev")) { into("data/config") }
-        from(layout.projectDirectory.file("../config.txt"))
-        from(layout.projectDirectory.file("../log4j2.xml"))
-        into(buildDir)
-        shouldRunAfter(tasks.named("copyNodeData"))
-    }
-
 // Copy dependencies into `data/lib`
 val copyLib =
     tasks.register<Sync>("copyLib") {
@@ -183,18 +157,36 @@ val copyApp =
         shouldRunAfter(tasks.named("copyLib"))
     }
 
+// Working directory for 'run' tasks
+val nodeWorkingDir = layout.buildDirectory.dir("node")
+
+val copyNodeData =
+    tasks.register<Sync>("copyNodeDataAndConfig") {
+        into(nodeWorkingDir)
+
+        // Copy things from hedera-node/data
+        into("lib") { from(copyLib) }
+        into("apps") { from(copyApp) }
+        into("onboard") { from(layout.projectDirectory.dir("../data/onboard")) }
+        into("data/keys") { from(layout.projectDirectory.dir("../data/keys")) }
+
+        // Copy hedera-node/configuration/dev as hedera-node/hedera-app/build/node/data/config  }
+        from(layout.projectDirectory.dir("../configuration/dev")) { into("data/config") }
+        from(layout.projectDirectory.file("../config.txt"))
+        from(layout.projectDirectory.file("../log4j2.xml"))
+    }
+
 tasks.assemble {
     dependsOn(copyLib)
     dependsOn(copyApp)
     dependsOn(copyNodeData)
-    dependsOn(copyConfig)
 }
 
 // Create the "run" task for running a Hedera consensus node
 tasks.register<JavaExec>("run") {
     group = "application"
     dependsOn(tasks.assemble)
-    workingDir = layout.projectDirectory.dir("./build/node").asFile
+    workingDir = nodeWorkingDir.get().asFile
     jvmArgs = listOf("-cp", "lib/*")
     mainClass.set("com.swirlds.platform.Browser")
 }
@@ -202,7 +194,7 @@ tasks.register<JavaExec>("run") {
 tasks.register<JavaExec>("modrun") {
     group = "application"
     dependsOn(tasks.assemble)
-    workingDir = layout.projectDirectory.dir("./build/node").asFile
+    workingDir = nodeWorkingDir.get().asFile
     jvmArgs = listOf("-cp", "lib/*:apps/*", "-Dhedera.workflows.enabled=true")
     mainClass.set("com.hedera.node.app.ServicesMain")
 }

--- a/hedera-node/hedera-app/build.gradle.kts
+++ b/hedera-node/hedera-app/build.gradle.kts
@@ -165,15 +165,16 @@ val copyNodeData =
         into(nodeWorkingDir)
 
         // Copy things from hedera-node/data
-        into("lib") { from(copyLib) }
-        into("apps") { from(copyApp) }
-        into("onboard") { from(layout.projectDirectory.dir("../data/onboard")) }
+        into("data/lib") { from(copyLib) }
+        into("data/apps") { from(copyApp) }
+        into("data/onboard") { from(layout.projectDirectory.dir("../data/onboard")) }
         into("data/keys") { from(layout.projectDirectory.dir("../data/keys")) }
 
         // Copy hedera-node/configuration/dev as hedera-node/hedera-app/build/node/data/config  }
         from(layout.projectDirectory.dir("../configuration/dev")) { into("data/config") }
         from(layout.projectDirectory.file("../config.txt"))
         from(layout.projectDirectory.file("../log4j2.xml"))
+        from(layout.projectDirectory.file("../configuration/dev/settings.txt"))
     }
 
 tasks.assemble {
@@ -187,7 +188,7 @@ tasks.register<JavaExec>("run") {
     group = "application"
     dependsOn(tasks.assemble)
     workingDir = nodeWorkingDir.get().asFile
-    jvmArgs = listOf("-cp", "lib/*")
+    jvmArgs = listOf("-cp", "data/lib/*")
     mainClass.set("com.swirlds.platform.Browser")
 }
 
@@ -195,7 +196,7 @@ tasks.register<JavaExec>("modrun") {
     group = "application"
     dependsOn(tasks.assemble)
     workingDir = nodeWorkingDir.get().asFile
-    jvmArgs = listOf("-cp", "lib/*:apps/*", "-Dhedera.workflows.enabled=true")
+    jvmArgs = listOf("-cp", "data/lib/*:data/apps/*", "-Dhedera.workflows.enabled=true")
     mainClass.set("com.hedera.node.app.ServicesMain")
 }
 

